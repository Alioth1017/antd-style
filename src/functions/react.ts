import _styled from '@emotion/styled';

import type { CreateStyled } from '@/types';

export const styled = _styled as CreateStyled;
export { css as reactCss, ThemeProvider as EmotionThemeProvider, withTheme } from '@emotion/react';
