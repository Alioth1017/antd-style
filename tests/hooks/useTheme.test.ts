import { renderHook } from '@testing-library/react';

import { ThemeProvider, useTheme } from 'antd-style';

describe('useTheme', () => {
  it('如果没有 Provider，theme 对象时是 antd Token 默认值', () => {
    const { result } = renderHook(useTheme);
    expect(result.current.colorPrimary.toLowerCase()).toEqual('#1677ff');
    expect(result.current.stylish).toEqual({});
  });

  it('包裹 Provider 后如果自定义主题变量', () => {
    const { result } = renderHook(useTheme, { wrapper: ThemeProvider });
    expect(result.current.colorPrimary.toLowerCase()).toEqual('#1677ff');
    expect(result.current.stylish).toEqual({});
  });

  it('自定义 Token', () => {});
  it('自定义 Stylish', () => {});
});
