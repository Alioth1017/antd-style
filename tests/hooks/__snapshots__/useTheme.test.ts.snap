// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTheme 包裹 Provider 后如果自定义主题变量，能拿到完整变量 1`] = `{}`;
